Library/Homebrew/system_command.rb

```diff
@@ -157,8 +163,9 @@ def env_args
 
   sig { returns(T::Array[String]) }
   def sudo_prefix
+    user_flags = sudo_as_root? ? ["-u", "root"] : []
```

Comment:
The ternary operator here makes this harder to read than it needs to be. Please either use a `if` here or use the way I suggested where you create an array and only add to it when `sudo_as_root?`. Please also do the same for `askpass_flags` below so it's consistent.